### ISSUE TITLE
QUESTION: How to connect to a remote database via SSH?

### ISSUE THREAD

**Issue by kigathi-chege** [dbgate/dbgate#714]
**Details:**
I have a remote server, which I am able to access through ssh. I want to be able to access my mysql database using dbgate. If I use the general connection, and use my server's IP as the server, the connection times out (ETIMEDOUT) before it can be established. On the other hand, I don't know how to use the SSH Tunnel, which host, bastion host, ssh authentication, and login, or any other details, should I use, and how? 

**Version Information [might be relevant to your issue]**
Operating System: Ubunt, Linux
App Version [help -> about]: 5.2.7

**Screenshot [if appropriate]**:
A screenshot of the app if that helps
![image](https://github.com/dbgate/dbgate/assets/30687709/bedc6518-8e29-4062-bff9-758742a95628)

**Comment by ProjectInfinity** [comment]
Hi.

In order to connect to a database through a SSH tunnel, you would need to use localhost (or the IP of the server if the tunnel is remote) in the general tab.

I am going to assume you would like to SSH tunnel to the same server that the MySQL server runs on.

In general, you would put your credentials and the `Server` field should say `localhost`.
Next on the SSH tunnel tab put the IP of the server that you wish to use as a middleman, in this case I will assume it is the server IP you put in the general tab before you changed it to localhost. Next fill out your SSH information as if you were running a SSH command in your terminal. Standard SSH port is 22, then add username and password or whatever authentication method you use.

**Comment by kigathi-chege** [comment]
This is how I am doing it:

![image](https://github.com/dbgate/dbgate/assets/30687709/31f5c832-f595-436a-90b5-b6ec89a59872)

and

![image](https://github.com/dbgate/dbgate/assets/30687709/e4615f72-34c2-48eb-aafb-c8f9c15c7412)

Still getting 

![image](https://github.com/dbgate/dbgate/assets/30687709/8e761b35-5134-4020-a9f8-683d085c57f3)

I am however able to still ssh into my server, and even create an ssh tunnel via terminal.

Am I missing anything?

**Comment by ProjectInfinity** [comment]
Do you happen to be on a IPv6 network?

**Comment by kigathi-chege** [comment]
My remote server is configured for IPV6, but my local server is configured primarily for IPV4, with some link-local IPV4 addresses present for local network communication.

**Comment by cicciopizza** [comment]
I have the same problem: 
I try to make a connection with ssh tunnel and I get this error:

![Selezione_022](https://github.com/dbgate/dbgate/assets/26459829/e7115fad-e0dc-44c2-8242-3792f4eabcf0)

The exact same SSH tunnel connection setup done with dbeaver-ce from the same PC works perfectly !!!

**Comment by kigathi-chege** [comment]
@cicciopizza 

I was able to successfully work around the issue by creating a remote user and connecting normally via the server IP.

Check out
 [Tech Republic](https://www.techrepublic.com/article/create-mysql-8-database-user-remote-access-databases/)

**Comment by cicciopizza** [comment]
@kigathi-chege

Yes, using ssh user and password works for me too, the problem is that I have several remote databases which I access with ssh keys and there are no ssh users defined... !!!

In my opinion there must be some bug, let's hope they solve it because otherwise it isn't usable for me and I will have to continue using dbeaver-ce (which is "very heavy" and not intuitive... I don't like it at all)

**Comment by huypn** [comment]
Same issue for me. hope this issue will be fixed soon.

**Comment by ProjectInfinity** [comment]
Please try the following Beta https://github.com/dbgate/dbgate/releases/tag/v5.2.9-beta.1

**Comment by techtanic** [comment]
This is still an issue in 5.4.4
